File: Selenium/Project/Project_Java/Activity6.java
package Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class Activity6 {
    public static void main(String[] args) {
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://alchemy.hguy.co/jobs/jobs/");

        // Search for a particular job (e.g., "Automation Engineer")
        WebElement searchJob = driver.findElement(By.id("search_keywords"));
        searchJob.sendKeys("Automation Engineer");
        searchJob.submit();

        // Wait for the job listings to show
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        wait.until(ExpectedConditions.presenceOfElementLocated(By.className("job_listing")));

        // Click and open any one of the jobs listed
        WebElement jobListing = driver.findElement(By.xpath("//ul[@class='job_listings']"));
        jobListing.click();

        // Click the apply button
        WebElement applyButton = driver.findElement(By.xpath("//input[@value='Apply for job']"));
        applyButton.click();

        // Print the email to the console
        WebElement emailElement = driver.findElement(By.cssSelector(".job_application_email"));
        String email = emailElement.getText();
        System.out.println("Job application email: " + email);

        // Close the browser
        driver.quit();
    }
}

File: Selenium/Project/Python_Project/Activity6.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

# Initialize the WebDriver
driver = webdriver.Chrome()

try:
    # Open browser with Alchemy Jobs site and navigate to the Jobs page
    driver.get('https://alchemy.hguy.co/jobs/jobs/')
    
    # Search for a particular job
    search_job = driver.find_element(By.ID,"search_keywords")
    search_job.send_keys("Automation Engineer")  # You can change this to the job you want to search for
    search_job.submit()
    
    # Wait for listings to show
    WebDriverWait(driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, "job_listing")))
    
    # Click and open any one of the jobs listed
    job_link = driver.find_element("xpath","//ul[@class='job_listings']")
    job_link.click()
    
    # Click the apply button
    apply_button = driver.find_element(By.CSS_SELECTOR,".application_button")
    apply_button.click()
    
    # Print the email to the console
    email = driver.find_element(By.CSS_SELECTOR,".job_application_email").text
    print("Job application email:", email)

finally:
    # Close the browser
    driver.quit()
